dispatch/static/manager/src/js/vendor/dispatch-editor/embeds/TwitterEmbed.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+
+import fetchJsonp from 'fetch-jsonp'
+import url from 'url'
+
+import { Button } from '@blueprintjs/core'
+import { FormInput, TextInput } from '../../../components/inputs'
+
+require('isomorphic-fetch')
+require('../styles/embeds/video.scss')
+
+const TWITTER_API_URL = 'https://api.twitter.com/1.1/statuses/oembed.json'
+
+class TwitterEmbedComponent extends React.Component {
+
+  validateTweetURL(url) {
+    return /^(?:https?:\/\/)?(?:www\.)?twitter\.com\/?(\w+)\/(status)\/(\d+)/g.test(url)
+  }
+
+  getTweet() {
+    const that = this
+    if (this.validateTweetURL(this.props.data.url)) {
+      fetchJsonp(TWITTER_API_URL + url.format({query: {url: this.props.data.url}}))
+      .then((response) => response.json())
+      .then(data => {
+        that.props.updateField('tweet', data.html)
+        this.props.stopEditing()
+      })
+    }
+  }
+
+  renderInput() {
+    return (
+      <div className='o-embed o-embed--tweet'>
+        <form>
+          <FormInput label='URL'>
+            <TextInput
+              fill={true}
+              value={this.props.data.url}
+              onChange={e => this.props.updateField('url', e.target.value)} />
+          </FormInput>
+        </form>
+        <div className='o-embed--video__button__container'>
+          <Button onClick={() => this.getTweet()}>Insert</Button>
+        </div>
+      </div>
+    )
+  }
+
+  renderTweet() {
+    return (
+      <div>
+        <div dangerouslySetInnerHTML={{__html: `${this.props.data.tweet}`}} />
+      </div>
+    )
+  }
+
+  render() {
+    return (
+      <div className='o-embed o-embed--tweet'>
+        {this.props.data.tweet ? this.renderTweet() : this.renderInput()}
+      </div>
+    )
+  }
+}
+
+export default {
+  type: 'tweet',
+  component: TwitterEmbedComponent,
+  defaultData: {
+    url: '',
+    id: '',
+    tweet: ''
```

Comment:
This is nitpicky, but I think it might make more sense for this field to be named `html`.